support-frontend/app/admin/settings/LandingPageTest.scala

```diff
@@ -0,0 +1,92 @@
+package admin.settings
+
+import com.gu.support.encoding.Codec
+import com.gu.support.encoding.Codec.deriveCodec
+import io.circe.generic.extras.semiauto.{deriveEnumerationDecoder, deriveEnumerationEncoder}
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.{Decoder, Encoder}
+
+sealed trait Status
+object Status {
+  case object Live extends Status
+  case object Draft extends Status
+
+  implicit val statusEncoder = deriveEnumerationEncoder[Status]
+  implicit val statusDecoder = deriveEnumerationDecoder[Status]
+}
+
+case class LandingPageTestTargeting(
+    countryGroups: List[String],
+)
+
+object LandingPageTestTargeting {
+  implicit val codec: Codec[LandingPageTestTargeting] = deriveCodec
```

Comment:
I don't think seen (or noticed) Codec before - what does it do?

Reply:
It's a [class](https://github.com/guardian/support-frontend/blob/main/support-models/src/main/scala/com/gu/support/encoding/Codec.scala#L12) in this repo which groups together Circe's concepts of Encoders + Decoders for handling JSON.
Here we're using implicit magic to generate the necessary JSON logic